node/src/components/storage.rs

```diff
@@ -712,6 +751,37 @@ impl Storage {
             .transpose()
     }
 
+    pub fn get_highest_block_header_for_db_check(&self) -> Option<Block> {
+        let mut txn = self.env.begin_ro_txn().ok()?;
+        if let Some(height) = self.block_height_index.keys().last() {
+            self.get_block_by_height(&mut txn, *height).ok()?
+        } else {
+            None
+        }
+    }
+
+    pub fn get_state_root_hashes_for_trie_check(&self) -> Option<Vec<Blake2bHash>> {
+        let mut blake_hashes: Vec<Blake2bHash> = Vec::new();
+        let header_txn = match self.env.begin_ro_txn() {
+            Ok(txn) => txn,
+            Err(_) => return None,
+        };
+        let mut cursor = match header_txn.open_ro_cursor(self.block_header_db) {
+            Ok(cursor) => cursor,
+            Err(_) => return None,
+        };
```

Comment:
You can replace the `match` statements here with `.ok()?`. `ok()` turns `Result` into `Option` type, ignoring the error.

Reply:
Removed match statements